src/app/modules/verification/__tests__/verification.routes.spec.ts

```diff
@@ -0,0 +1,359 @@
+      expect(response.body).toEqual(
+        buildCelebrateError({ body: { key: 'formId' } }),
+      )
+    })
+
+    it('should return 400 when formId is malformed', async () => {
+      const response = await request.post(ROUTE).send({ formId: 'malformed' })
+
+      expect(response.status).toBe(StatusCodes.BAD_REQUEST)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'formId',
+            message: '"formId" length must be 24 characters long',
+          },
+        }),
+      )
+    })
+
+    it('should return 200 when form has no verifiable fields', async () => {
+      const response = await request
+        .post(ROUTE)
+        // ID of form with no verifiable fields
+        .send({ formId: mockEmptyFormId })
+
+      expect(response.status).toBe(StatusCodes.OK)
+      expect(response.body).toEqual({})
+    })
+
+    it('should return 201 when form has verifiable fields', async () => {
+      const response = await request
+        .post(ROUTE)
+        // ID of form with verifiable fields
+        .send({ formId: mockVerifiableFormId })
+
+      expect(response.status).toBe(StatusCodes.CREATED)
+      expect(response.body).toEqual({
+        transactionId: expect.any(String),
+        expireAt: expect.any(String),
+      })
+    })
+  })
+
+  describe('GET /:transactionId', () => {
+    it('should return 200 when transactionId is valid', async () => {
+      const response = await request.get(`/transaction/${mockTransactionId}`)
+
+      expect(response.status).toBe(StatusCodes.OK)
+      expect(response.body).toEqual({
+        formId: expect.any(String),
+        expireAt: expect.any(String),
+        _id: expect.any(String),
+      })
+    })
+  })
+
+  describe('POST /:transactionId/reset', () => {
+    it('should return 400 when fieldId is not provided in body', async () => {
+      const response = await request.post(
+        `/transaction/${mockTransactionId}/reset`,
+      )
+
+      expect(response.status).toBe(StatusCodes.BAD_REQUEST)
+      expect(response.body).toEqual(
+        buildCelebrateError({ body: { key: 'fieldId' } }),
+      )
+    })
+
+    it('should return 400 when fieldId is malformed', async () => {
+      const response = await request
+        .post(`/transaction/${mockTransactionId}/reset`)
+        .send({ fieldId: 'malformed' })
+
+      expect(response.status).toBe(StatusCodes.BAD_REQUEST)
+      expect(response.body).toEqual(
+        buildCelebrateError({
+          body: {
+            key: 'fieldId',
+            message: '"fieldId" length must be 24 characters long',
+          },
+        }),
+      )
+    })
+
+    it('should return 200 when transactionId and fieldId for email field are valid', async () => {
+      const response = await request
+        .post(`/transaction/${mockTransactionId}/reset`)
+        .send({ fieldId: mockEmailFieldId })
+
+      expect(response.status).toBe(StatusCodes.OK)
```

Comment:
do you think we should verify the shape of the body also? can use `http-status-code`'s `getReasonPhrase`

Reply:
sure